Presentation/CustomAdaptivePresentationDelegate.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  CustomAdaptivePresentationDelegate.swift
+//  Presentation
+//
+//  Created by Vasil Blanco-Nunev on 2019-07-19.
+//  Copyright © 2019 iZettle. All rights reserved.
+//
+
+import UIKit
+import Flow
+
+public class CustomAdaptivePresentationDelegate: NSObject, UIAdaptivePresentationControllerDelegate {
+    private let didAttemptToDismissCallbacker = Callbacker<()>()
+
+    /// A signal that fires when a modally presented view controller is attempting to be dismissed from a swipe down action
+    ///
+    /// The view will attempt to dismiss only when the  view controller's`.isModalInPresentation` is set to true
+    public var didAttemptToDismissSignal: Signal<()> {
+        return Signal(callbacker: didAttemptToDismissCallbacker)
+    }
+
+    @available(iOS 13.0, *)
```

Comment:
Would it make sense to move this to the class itself to get the appearance that it will be useful older iOS versions?